evalml/model_understanding/graphs.py

```diff
@@ -100,29 +100,36 @@ def graph_confusion_matrix(y_true, y_pred, normalize_method='true', title_additi
         plotly.Figure representing the confusion matrix plot generated
     """
     _go = import_or_raise("plotly.graph_objects", error_msg="Cannot find dependency plotly.graph_objects")
+    _ff = import_or_raise("plotly.figure_factory", error_msg="Cannot find dependency plotly.figure_factory")
     if jupyter_check():
         import_or_raise("ipywidgets", warning=True)
 
     conf_mat = confusion_matrix(y_true, y_pred, normalize_method=None)
     conf_mat_normalized = confusion_matrix(y_true, y_pred, normalize_method=normalize_method or 'true')
-    labels = conf_mat.columns
+    labels = conf_mat.columns.tolist()
 
     title = 'Confusion matrix{}{}'.format(
         '' if title_addition is None else (' ' + title_addition),
         '' if normalize_method is None else (', normalized using method "' + normalize_method + '"'))
     z_data, custom_data = (conf_mat, conf_mat_normalized) if normalize_method is None else (conf_mat_normalized, conf_mat)
+    z_data = z_data.to_numpy()
+    z_text = [["{:.3f}".format(y) for y in x] for x in z_data]
     primary_heading, secondary_heading = ('Raw', 'Normalized') if normalize_method is None else ('Normalized', 'Raw')
     hover_text = '<br><b>' + primary_heading + ' Count</b>: %{z}<br><b>' + secondary_heading + ' Count</b>: %{customdata} <br>'
     # the "<extra> tags at the end are necessary to remove unwanted trace info
     hover_template = '<b>True</b>: %{y}<br><b>Predicted</b>: %{x}' + hover_text + '<extra></extra>'
     layout = _go.Layout(title={'text': title},
                         xaxis={'title': 'Predicted Label', 'type': 'category', 'tickvals': labels},
                         yaxis={'title': 'True Label', 'type': 'category', 'tickvals': labels})
-    fig = _go.Figure(data=_go.Heatmap(x=labels, y=labels, z=z_data,
-                                      customdata=custom_data,
-                                      hovertemplate=hover_template,
-                                      colorscale='Blues'),
-                     layout=layout)
+    fig = _ff.create_annotated_heatmap(z_data, x=labels, y=labels,
+                                       annotation_text=z_text,
+                                       customdata=custom_data,
+                                       hovertemplate=hover_template,
+                                       colorscale='Blues',
+                                       showscale=True)
+    fig.update_layout(layout)
+    # put xaxis text on bottom to not overlap with title
```

Comment:
This is still showing up on the top and overlapping to me:
![image](https://user-images.githubusercontent.com/5422235/103585807-c8349800-4eb1-11eb-928f-849d0b42b9f6.png)

Reply:
Commented on below